src/DotNetWorker.Core/DotNetWorker.Core.csproj

```diff
@@ -22,6 +21,16 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.41.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Collections.Immutable" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="9.0.0-preview.*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
```

Comment:
If we're adding the .NET 8 target with this PR, should we also do the same thing for .NET 8 here?

Reply:
So, I updated the PR to pick only those 2 commits. The changes you suggested can go as a separate commit.

Reply:
@kshyju will be following up with changes against 2.x